app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,26 @@
+class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def google_oauth2
```

Comment:
Repeating yourself in these two methods. Maybe a centralized `do_omniauth` method accepting the kind & session key to append to?